src/main/java/org/hashids/Hashids.java

```diff
@@ -228,150 +243,154 @@ private String _encode(long... numbers) {
     for (int i = 0; i < numbers.length; i++) {
       numberHashInt += (numbers[i] % (i + 100));
     }
-    String alphabet = this.alphabet;
-    final char ret = alphabet.charAt((int) (numberHashInt % alphabet.length()));
+
+    char[] newAlphabet = alphabet;
+    final char ret = newAlphabet[(int) (numberHashInt % newAlphabet.length)];
 
     long num;
     long sepsIndex, guardIndex;
-    String buffer;
-    final StringBuilder ret_strB = new StringBuilder(this.minHashLength);
+    final StringBuilder buffer = new StringBuilder();
+    final StringBuilder ret_strB = new StringBuilder();
     ret_strB.append(ret);
     char guard;
 
     for (int i = 0; i < numbers.length; i++) {
       num = numbers[i];
-      buffer = ret + this.salt + alphabet;
+      buffer.setLength(0);
+      buffer.append(ret)
+          .append(salt)
+          .append(newAlphabet);
 
-      alphabet = Hashids.consistentShuffle(alphabet, buffer.substring(0, alphabet.length()));
-      final String last = Hashids.hash(num, alphabet);
+      newAlphabet = Hashids.consistentShuffle(newAlphabet, buffer.substring(0, newAlphabet.length).toCharArray());
+      final String last = Hashids.hash(num, newAlphabet);
 
       ret_strB.append(last);
 
       if (i + 1 < numbers.length) {
         if (last.length() > 0) {
-          num %= (last.charAt(0) + i);
-          sepsIndex = (int) (num % this.seps.length());
+          num %= last.charAt(0) + i;
+          sepsIndex = (int) (num % seps.length);
         } else {
           sepsIndex = 0;
         }
-        ret_strB.append(this.seps.charAt((int) sepsIndex));
+
+        ret_strB.append(seps[(int) sepsIndex]);
       }
     }
 
-    String ret_str = ret_strB.toString();
-    if (ret_str.length() < this.minHashLength) {
-      guardIndex = (numberHashInt + (ret_str.charAt(0))) % this.guards.length();
-      guard = this.guards.charAt((int) guardIndex);
+    if (ret_strB.length() < minHashLength) {
+      guardIndex = (numberHashInt + (ret_strB.charAt(0))) % guards.length;
+      guard = guards[(int) guardIndex];
 
-      ret_str = guard + ret_str;
+      ret_strB.insert(0, guard);
 
-      if (ret_str.length() < this.minHashLength) {
-        guardIndex = (numberHashInt + (ret_str.charAt(2))) % this.guards.length();
-        guard = this.guards.charAt((int) guardIndex);
+      if (ret_strB.length() < minHashLength) {
```

Comment:
always prefix methods/attributes with `this.`

Reply:
Why? This is not a JAVA rule. Perhaps a JS one.

Reply:
This is not a java rule, this is our convention rule, all call to instance members are prefixed with `this.` Read our code.

Reply:
It is pretty hard to add your contribution to the community if every project has its standards. This is why there are Java code conventions (by Sun, Oracle, Google - which are quite similar).